### PR TITLE
New Chem: Rubbing Alcohol

### PR DESCRIPTION
**Pull request by Wisemonster**
[FEEDBACK]

## About the PR 
Adds a new chemical: Isopropanol! This chem is made by combining 1:hydrogen,1:oxygen, and 2:ethanol. This also replaces the normal ethanol in the rubbing alcohol bottles with this new chem.

## Why's this needed? 
The main purpose of this chem is to replace the ethanol in rubbing alcohol bottles with a chem that actually does something. This is meant to reduce the damage dealt when performing surgery, and also decays into ethanol.

## Changelog

```changelog
(u)Wisemonster
(*)Added a new chem, isopropanol! (AKA, rubbing alcohol) Check minor changes for details
(+)Isopropanol is a chem that's helpful in surgery. It's components are hydrogen, oxygen, and ethanol, with a 2:1 ratio for ethanol. More info in the Pharmacopia.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
I'd prefer a more descriptive name than just "rubbing alcohol". Rubbing alcohol can be ethanol(plus things that make it unpleasant to drink) or isopropanol . Seems like you're going for isopropanol given its recipe is just ethanol + oxygen and hydrogen.

**Comment by Nexusuxen** [comment]
A minor suggestion, but it'd be kinda neat if this also increased your hygiene motive on the RP servers on account of you disinfecting your body.

Otherwise a nifty addition!

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -2270,6 +2270,14 @@ datum
 			result_amount = 4
 			mix_phrase = "The solution yields an astringent powder."
 
+		rubbing_alcohol
+			name = "Rubbing Alcohol"
```
Comment (Zonespace27) [inline code/modules/chemistry/Chemistry-Recipes.dm:2274]: might be just me but the chem name (isopropanol) and the name of the recipe should probably be the same

File: code/modules/chemistry/Reagents-Medical.dm [hunk 2]
```diff
@@ -1255,6 +1255,50 @@ datum
...
+					. = 0
+					M.HealDamage("All", volume_passed/2, 0)
+
+					var/silent = 0
+					if (length(paramslist))
+						if ("silent" in paramslist)
+							silent = 1
+
+					if (!silent)
+						if(!ON_COOLDOWN(M, "rubbing alcohol gasp", 3 SECONDS))
```
Comment (Zonespace27) [inline code/modules/chemistry/Reagents-Medical.dm:1295]:
```suggestion
						if(!ON_COOLDOWN(M, "rubbing_alcohol_gasp", 3 SECONDS))
```
generally spaces aren't used

File: code/modules/chemistry/Reagents-Medical.dm [hunk 3]
```diff
@@ -1255,6 +1255,50 @@ datum
...
+					return
+				if(!isliving(M))
+					return
+				volume_passed = clamp(volume_passed, 0, 10)
+
+				if(method == TOUCH)
+					. = 0
+					M.HealDamage("All", volume_passed/2, 0)
+
+					var/silent = 0
```
Comment (Zonespace27) [inline code/modules/chemistry/Reagents-Medical.dm:1289]:
```suggestion
					var/silent = FALSE
```

File: code/modules/chemistry/Reagents-Medical.dm [hunk 4]
```diff
@@ -1255,6 +1255,50 @@ datum
...
+					if (length(paramslist))
+						if ("silent" in paramslist)
+							silent = 1
```
Comment (Zonespace27) [inline code/modules/chemistry/Reagents-Medical.dm:1292]:
```suggestion
							silent = TRUE
```

**Comment by ThatFiveGuys** [comment]
Maybe add it to automender whitelist?

**Comment by Wisemonster** [comment]
Wasn't sure how text for the pharmacopia works, but I added it the best I could.

**Comment by Yellow-Mushroom** [comment]
Could this also take off stickers? Other than keeping skin clean, this is the other main use of rubbing alcohol

**Comment by vocalpocal** [comment]
While I think it is long overdue to have rubbing alcohol, I don't think brute healing fits to it thematically. If we had more complex skinpen mechanics, this could be the perfect answer to slowly absorbing chems.
Also if you want to make it so that people get drunk by drinking this, you could make isopropanol a child of ethanol, and then add your own toxic effects on ingestion... I think?

**Review by Tarmunora — CHANGES_REQUESTED** [review]
I think that overall, I'd rather have this do something new/unique as opposed to being more-or-less a styptic reskin.
Maybe something more along the lines of giving a status that reduces surgery fail chance, surgery fail rates, or successful surgery damage? A "prep the patient" chem, rather than a "lather them in rubbing alcohol after cutting them open" kind of thing

**Comment by Wisemonster** [comment]
> 
> 
> I think that overall, I'd rather have this do something new/unique as opposed to being more-or-less a styptic reskin. Maybe something more along the lines of giving a status that reduces surgery fail chance, surgery fail rates, or successful surgery damage? A "prep the patient" chem, rather than a "lather them in rubbing alcohol after cutting them open" kind of thing

Aight, I'll work on giving it some kinda surgery utility tommorow.

**Comment by Kyle2143** [comment]
I don't think that giving isopropanol a surgery disinfectant type application like that really makes sense unless such an application applies to other chems that are also used as disinfectants. I'd think that iodine would be the go to in that case.

File: code/modules/medical/surgery_procs.dm [hunk 5]
```diff
@@ -329,8 +329,11 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
 	DEBUG_MESSAGE("<b>[patient]'s surgery (performed by [surgeon]) damage_low is [damage_low], damage_high is [damage_high]</b>")
 
 /* ---------- SCALPEL - HEAD ---------- */
-
 	if (surgeon.zone_sel.selecting == "head")
+		if (patient.hasStatus("sterilized")) // reduce the rand value by 5 up and down, effectivily reducing it by 10
+			damage_low = calc_surgery_damage(surgeon, screw_up_prob, rand(0,10) * surgCheck/*, src.adj1, src.adj2*/)
+			damage_high = calc_surgery_damage(surgeon, screw_up_prob, rand(10,20) * surgCheck/*, src.adj1, src.adj2*/)
+
```
Comment (Tarmunora) [inline code/modules/medical/surgery_procs.dm:336]: This seems like a very messy way of going about this. Please consider a different approach :/

File: code/modules/status_system/statusEffects.dm [hunk 6]
```diff
@@ -2018,3 +2018,10 @@
 			APC.lighting = oldstate
 			APC.updateicon()
 			APC.update()
+
+/datum/statusEffect/sterilized
```
Comment (Tarmunora) [inline code/modules/status_system/statusEffects.dm:2022]: This status arguably doesn't need to be visible at all

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Reagents-Medical.dm [hunk 7]
```diff
@@ -1255,6 +1255,93 @@ datum
...
+			fluid_b = 0
+			transparency = 10
+			value = 2 //Not sure what to set this, so it'll just be low
+			var/alch_strength = 0.6
+			hygiene_value = 1
+
+			on_mob_life(var/mob/M, var/mult = 1)
+				if(!M) M = holder.my_atom
+				M.reagents.add_reagent("ethanol", alch_strength * mult)
+				M.reagents.remove_reagent(src, 1 * mult)
```
[the remark this answers is not in the document]
Comment (Wisemonster) [reply code/modules/chemistry/Reagents-Medical.dm:1274]: This is meant to make the chem alchoholic. Without it, it didn't deplete into ethanol.

File: code/modules/chemistry/Chemistry-Reactions.dm [hunk 8]
```diff
@@ -211,3 +211,28 @@
...
+				removed_count++
+		if (removed_count > can_remove_amt)
+			break
+
+/proc/locate_sticker_wall(var/turf/T)
+	for (var/turf/turf in range(1,T))
+		for (var/obj/item/sticker/S in turf)
+			if (S.attached == T || S.attached.loc == T)
+				return T
+	return 0
```
Comment (Tarmunora) [inline code/modules/chemistry/Chemistry-Reactions.dm:238]: acetone should use this as well

File: code/modules/chemistry/Reagents-Medical.dm (hunk 7 above)
[the remark this answers is not in the document]
Comment (Tarmunora) [reply code/modules/chemistry/Reagents-Medical.dm:1278]: chem depletion is handled in on_mob_life parent, you don't need to roll your own in here

File: code/modules/chemistry/tools/hyposprays.dm [hunk 9]
```diff
@@ -1,6 +1,6 @@
 var/global/list/chem_whitelist = list("antihol", "charcoal", "epinephrine", "insulin", "mutadone", "teporone",\
 "silver_sulfadiazine", "salbutamol", "perfluorodecalin", "omnizine", "synaptizine", "anti_rad",\
-"oculine", "mannitol", "penteticacid", "styptic_powder", "methamphetamine", "spaceacillin", "saline",\
+"oculine", "mannitol", "penteticacid", "styptic_powder", "methamphetamine", "spaceacillin", "saline", "isopropanol",\
```
Comment (Tarmunora) [inline code/modules/chemistry/tools/hyposprays.dm:3]: don't think it's a good fit for the medical whitelist

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/tools/hyposprays.dm (hunk 9 above)
Comment (Wisemonster) [reply code/modules/chemistry/tools/hyposprays.dm:3]: Fair enough, forgot to remove it after removing the brute healing effect.

**Pull request closed without merging**